r-package/grf/R/regression_forest.R

```diff
@@ -238,6 +253,7 @@ predict.regression_forest <- function(object, newdata = NULL,
             training.data <- create_data_matrices(X.orig)
             ret <- local_linear_predict(forest.short, data$default, training.data$default, data$sparse,
                 training.data$sparse, lambda, use_unweighted_penalty, linear.correction.variables, num.threads)
+            return(ret)
```

Comment:
I wasn't able to reproduce the issue here. Is there an example where the below line `do.call(...)` doesn't work, as you mentioned over chat?

Reply:
Okay, I tracked down the issue: in `tune_local_linear_forest, you're doing `preds.matrix = predict(forest, linear.correction.variables = linear.correction.variables, lambda = lambdas)$predictions`. However, if multiple values for `lambda` were provided, then there is no `predictions` key, but rather `predictions.1`, `predictions.2`, etc. I think that it'd be cleaner to call into `local_linear_predict` here directly, so that we maintain the invariant that the top-level `predict` only ever returns on set of predictions (and not a full regularization path).

The fact that this was working at all also suggests the smoke test may not be strong enough.

Reply:
Hey Julie- the only caveat is that if we maintain the top-level predict returning only one set of predictions, it's not clear to me how we'd let users efficiently compute a full regularization path. We could add a different type of key, maybe- as in, if predictions has 1 column it's accessed by $predictions, and otherwise by $predictionpath or something? That's not the cleanest though, so let me know what you're thinking.

Reply:
I think that this will work out okay given my other suggestion (that we should return a full regularization path in `tune_local_linear_forests`, and have the user pass in the `lambda` they want to `predict`).

Reply:
Okay I think that makes sense! Looking at it now :)